File: ios/CryptoSavingsTracker/Views/Planning/FixedBudgetIntroCard.swift
//
//  FixedBudgetIntroCard.swift
//  CryptoSavingsTracker
//
//  Introduces Fixed Budget Mode to users who haven't seen it yet
//

import SwiftUI

/// A promotional card introducing Fixed Budget Mode to users
struct FixedBudgetIntroCard: View {
    let onLearnMore: () -> Void
    let onTryIt: () -> Void
    let onDismiss: () -> Void

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            // Header with dismiss button
            HStack {
                Image(systemName: "sparkles")
                    .foregroundStyle(.yellow)
                    .font(.title3)

                Text("New: Fixed Budget Mode")
                    .font(.headline)
                    .foregroundStyle(.primary)

                Spacer()

                Button(action: onDismiss) {
                    Image(systemName: "xmark.circle.fill")
                        .foregroundStyle(.secondary)
                        .font(.title3)
                }
                .buttonStyle(.plain)
                .accessibilityLabel("Dismiss intro card")
            }

            // Description
            Text("Set one monthly savings amount, and we'll optimize which goals to fund first based on deadlines.")
                .font(.subheadline)
                .foregroundStyle(.primary)

            Text("Perfect for users with fixed salaries who want predictable monthly payments.")
                .font(.caption)
                .foregroundStyle(.secondary)

            // Action buttons
            HStack {
                Button("Learn More", action: onLearnMore)
                    .font(.subheadline)
                    .foregroundStyle(AccessibleColors.primaryInteractive)

                Spacer()

                Button(action: onTryIt) {
                    Text("Try It")
                        .font(.subheadline)
                        .fontWeight(.semibold)
                }
                .buttonStyle(.borderedProminent)
            }
            .padding(.top, 4)
        }
        .padding()
        #if os(iOS)
        .background(Color(.secondarySystemGroupedBackground))
        #else
        .background(Color.gray.opacity(0.1))
        #endif
        .clipShape(RoundedRectangle(cornerRadius: 12))
        .overlay(
            RoundedRectangle(cornerRadius: 12)
                .stroke(Color.yellow.opacity(0.3), lineWidth: 1)
        )
    }
}

// MARK: - Preview

#Preview {
    VStack {
        FixedBudgetIntroCard(
            onLearnMore: {},
            onTryIt: {},
            onDismiss: {}
        )
        .padding()
    }
    .background(Color.gray.opacity(0.1))
}

File: android/app/src/main/java/com/xax/CryptoSavingsTracker/presentation/planning/components/FixedBudgetIntroCard.kt
package com.xax.CryptoSavingsTracker.presentation.planning.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AutoAwesome
import androidx.compose.material.icons.filled.Close
import androidx.compose.material3.Button
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.xax.CryptoSavingsTracker.presentation.theme.CryptoSavingsTrackerTheme

/**
 * A promotional card introducing Fixed Budget Mode to users who haven't seen it yet.
 */
@Composable
fun FixedBudgetIntroCard(
    onLearnMore: () -> Unit,
    onTryIt: () -> Unit,
    onDismiss: () -> Unit,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        shape = RoundedCornerShape(12.dp),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surfaceVariant
        ),
        border = BorderStroke(1.dp, Color(0xFFFFD700).copy(alpha = 0.3f))
    ) {
        Column(
            modifier = Modifier.padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            // Header with dismiss button
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Row(
                    horizontalArrangement = Arrangement.spacedBy(8.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        imageVector = Icons.Default.AutoAwesome,
                        contentDescription = null,
                        tint = Color(0xFFFFD700),
                        modifier = Modifier.size(24.dp)
                    )
                    Text(
                        text = "New: Fixed Budget Mode",
                        style = MaterialTheme.typography.titleMedium,
                        fontWeight = FontWeight.SemiBold
                    )
                }

                IconButton(
                    onClick = onDismiss,
                    modifier = Modifier.size(24.dp)
                ) {
                    Icon(
                        imageVector = Icons.Default.Close,
                        contentDescription = "Dismiss intro card",
                        tint = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            }

            // Description
            Text(
                text = "Set one monthly savings amount, and we'll optimize which goals to fund first based on deadlines.",
                style = MaterialTheme.typography.bodyMedium
            )

            Text(
                text = "Perfect for users with fixed salaries who want predictable monthly payments.",
                style = MaterialTheme.typography.bodySmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )

            Spacer(modifier = Modifier.height(4.dp))

            // Action buttons
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                TextButton(onClick = onLearnMore) {
                    Text("Learn More")
                }

                Button(onClick = onTryIt) {
                    Text("Try It")
                }
            }
        }
    }
}

@Preview(showBackground = true)
@Composable
private fun FixedBudgetIntroCardPreview() {
    CryptoSavingsTrackerTheme {
        FixedBudgetIntroCard(
            onLearnMore = {},
            onTryIt = {},
            onDismiss = {},
            modifier = Modifier.padding(16.dp)
        )
    }
}
